api/app/helpers/seeds_handler.rb

```diff
@@ -0,0 +1,259 @@
+            session.request("/api/v1#{path}", env)
+            return JSON.parse(mock_session.last_response.body, symbolize_names: true)
+        rescue JSON::ParserError
+            error = "#{action.to_s.upcase} #{path} is an invalid path."
+            return {status: 'error', message: error, payload: {}}
+        end
+    end
+
+    def read_json(file, log)
+        '''
+        Returns a JSON with the file content and a log containing messages if any
+
+        file: located relative to seed_data_dir
+        log: records error messages
+        '''
+        seed_data_dir = Rails.root.join('db', 'seed')
+        file_dir = seed_data_dir+file
+        begin
+            data = File.read(file_dir)
+            return JSON.parse(data, symbolize_names: true), log
+        rescue Errno::ENOENT => e
+            log.push(e)
+            return {}, log
+        rescue JSON::ParserError 
+            log.push("#{file_dir} is not a valid JSON file.")
+            return {}, log
+        end
+    end
+
+    def insert_data(chaining, file)
+        '''
+        Inserts JSON mock data into the routes defined in chaining.
+
+        chaining: an array with JSON as elements. Each element contains
+            get: route to get all entries for a table
+            create: route to create an entry in a specific table
+            unique: an array of attributes that makes an entry unique in a table
+        file: JSON file name containg seed data to be inserted
+        '''
+        log = []
+        empty_db, log = empty_records(chaining, log)
+        if empty_db
+            if file
+                json, log = read_json(file, log)
+                valid_templates, log = check_templates(json, log)
+                if valid_templates
+                    log = post_data(chaining, json, log)
+                end
+            else
+                log.push('No JSON file was given.')
+            end
+        end
+        puts log
+    end
+ 
+    def empty_records(chaining, log)
+        '''
+        Return whether or not the db is empty
+        '''
+        chaining.each do |entry, i|
+            res = request(:get, entry[:get])[:payload]
+            if res.length > 0
+                log.push('Database has not been cleared.')
+                return false, log
+            end
+        end
+        return true, log
+    end
+
+    def check_templates(json, log)
+        '''
+        Return whether or not the templates in the JSON file is valid
+        '''
+        templates = request(:get, '/available_position_templates')[:payload]
+        json[:position_templates].each do |entry|
+            data = {
+                offer_template: entry[:offer_template]
+            }
+            if not templates.include?(data)
+                log.push("Invalid position_template: #{entry[:offer_template]}")
+                return false, log
+            end
+        end
+        return true, log
+    end
+
+    def post_data(chaining, json, log)
```

Comment:
You're passing around `log` between several functions. This is a pattern `class`es were invented to solve. Would it make sense to refactor some of this into a class?

Reply:
I'm not sure which pattern class you're referring to. For now, I made use of `@log` to make it so that there's no passing around `log` through several functions.